src/client/clientlauncher.cpp

```diff
@@ -126,7 +128,8 @@ bool ClientLauncher::run(GameStartData &start_data, const Settings &cmd_args)
 		setAttribute(scene::ALLOW_ZWRITE_ON_TRANSPARENT, true);
 
 	guienv = m_rendering_engine->get_gui_env();
-	init_guienv(guienv);
+	config_guienv();
+	g_settings->registerChangedCallback("gui_scaling", setting_changed_callback);
```

Comment:
optional parameter missing which means `this == NULL` in `config_guienv` which is actually okay because it doesn't use any member variables

Reply:
```suggestion
	g_settings->registerChangedCallback("gui_scaling", setting_changed_callback, this);
```